File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        boolean answer = true;
        int a = 0;
        int size = s.length();
        
        for(int i=0; i<size; i++){            
            if(s.charAt(i) == '('){
                a++;
            }else{
                if(a == 0){
                    answer = false;
                    break;
                }
                a--;
                             
            }
            
        }    
        if(a> 0){
            answer = false;
        }
        return answer;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    
    answer = True
    
    stack = []
    
    for ch in s:
        if len(stack) == 0 and ch == ')':
            answer = False
            break
        if ch == '(':
            stack.append(ch)
        else:
            stack.pop()
    
    if len(stack) > 0:
        answer = False
    
    return answer